backend/src/controllers/stockController.ts

```diff
@@ -1,129 +1,254 @@
+//         const stockID = Number(req.params.stockID);
+//
+//         if (!itemID || QUANTITY === undefined)
+//             sendError(res, new ValidationError("ID and QUANTITY must be provided.", ErrorMessages.UpdateStockItemQuantity));
+//
+//         const stockService = new StockService(connection);
+//         const updateRequest = new UpdateStockRequest(itemID, QUANTITY, stockID);
+//         await stockService.updateStockItemQuantity(updateRequest);
+//         res.json({message: "Stock updated successfully."});
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
+//
+// export const addStockItem = async (req: Request, res: Response, connection: PoolConnection, stockID: number) => {
+//     try {
+//         const itemUpdated = {
+//             id: undefined,
+//             label: req.body['LABEL'],
+//             description: req.body['DESCRIPTION'],
+//             quantity: req.body['QUANTITY']
+//         };
+//         const stockService = new StockService(connection);
+//         await stockService.addStockItem(itemUpdated, stockID);
+//         res.status(201).json({message: "Stock item added successfully."});
+//     } catch (err: any) {
+//         if (err instanceof ValidationError) {
+//             sendError(res, err as CustomError);
+//         } else {
+//             sendError(res, err as CustomError);
+//         }
+//     }
+// };
+//
+// export const deleteStockItem = async (req: Request, res: Response, connection: PoolConnection, stockID: number, itemID: number) => {
+//     try {
+//         const stockService = new StockService(connection);
+//         const result = await stockService.deleteStockItem(stockID, itemID);
+//         if (result.affectedRows === 0)
+//             sendError(res, new NotFoundError("Item not found or already deleted.", ErrorMessages.DeleteStockItem));
+//         res.status(200).json({message: "Stock item deleted successfully."});
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
+//
+// export const getAllItems = async (req: Request, res: Response, connection: PoolConnection) => {
+//     try {
+//         const stockService = new StockService(connection);
+//         const items = await stockService.getAllItems();
+//         res.status(200).json(items);
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
+//
+// export const getItemDetails = async (req: Request, res: Response, connection: PoolConnection, itemID: number) => {
+//     try {
+//         const stockService = new StockService(connection);
+//         const items = await stockService.getItemDetails(itemID);
+//         if (items.length === 0)
+//             sendError(res, new NotFoundError("Item not found.", ErrorMessages.GetItemDetails));
+//         res.json(items[0]);
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
```

Comment:
issue(blocking): commented code to remove or uncomment.

Reply:
oui, en fait c'était pour voir quel est le mieux?